pkg/rust/cargo/parse.go

```diff
@@ -27,18 +31,57 @@ func NewParser() types.Parser {
 
 func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
 	var lockfile Lockfile
-	if _, err := toml.DecodeReader(r, &lockfile); err != nil {
+	decoder := toml.NewDecoder(r)
+	if _, err := decoder.Decode(&lockfile); err != nil {
 		return nil, nil, xerrors.Errorf("decode error: %w", err)
 	}
```

Comment:
we will parse lock file again(i mean `naivePkgParser`).
Perhaps it make sense to remove this decoding and take all dependency information from `naivePkgParser`?